### PR TITLE
feat(plugin-svelte): support pass preprocess options

### PR DESCRIPTION
**Pull request by tmkx**
## Summary

1. support pass preprocess options
2. add frequently used loader typings

## Checklist

- [x] Tests updated.
- [ ] Documentation updated.

### CODE REVIEW

File: packages/plugin-svelte/src/index.ts [hunk 1]
```diff
@@ -2,14 +2,40 @@ import path from 'path';
...
    * See https://github.com/sveltejs/svelte-loader
    */
-  svelteLoaderOptions?: Record<string, any>;
+  svelteLoaderOptions?: SvelteLoaderOptions;
+  /**
+   * The options of svelte-preprocess
+   *
+   * See https://github.com/sveltejs/svelte-preprocess
+   */
+  preprocessOptions?: AutoPreprocessOptions;
```
Comment (chenjiahan) [inline packages/plugin-svelte/src/index.ts:38]: Thanks! Can you add an unit test for this new option?

see: packages/plugin-svelte/tests/index.test.ts

Comment (tmkx) [reply packages/plugin-svelte/src/index.ts:39]: done :)

but the snapshot was not serialized expectably

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
Thanks! ❤️ 

Can you submit another PR to update the document?

File: packages/plugin-svelte/src/index.ts (hunk 1 above)
Comment (chenjiahan) [reply packages/plugin-svelte/src/index.ts:39]: haha, maybe we can improve the snapshot in the future

**Pull request merged**